app/renderer/js/utils/proxy-util.js

```diff
@@ -15,6 +16,39 @@ class ProxyUtil {
 		return instance;
 	}
 
+	// Return proxy to be used for a particular uri, to be used for request
+	getProxy(uri) {
+		uri = url.parse(uri);
+		const proxyRules = ConfigUtil.getConfigItem('proxyRules', '').split(';');
+		// If SPS is on and system uses no proxy then request should not try to use proxy from
+		// environment. NO_PROXY = '*' makes request ignore all environment proxy variables.
+		if (proxyRules[0] === '') {
+			process.env.NO_PROXY = '*';
+			return;
+		}
+
+		const proxyRule = {};
+		if (uri.protocol === 'http:') {
```

Comment:
@abhigyank we can refactor this function since it's the same for http + https, but let's leave it for now.